app/services/TwoStepVerification.scala

```diff
@@ -78,31 +78,35 @@ trait TwoStepVerification {
 
   def getDestinationVia2SV(continue: Location, ruleContext: RuleContext, auditContext: TAuditContext)(implicit authContext: AuthContext, request: Request[AnyContent], hc: HeaderCarrier) = {
 
-    def throttleLocation(rule: Biz2SVRule) = ruleContext.isAdmin.map(if (_) rule.adminLocations else rule.assistantLocations)
-
-    if (twoStepVerificationEnabled && continue == BusinessTaxAccount) {
+    def throttleLocations(biz2SVRule: Option[Biz2SVRule]) = biz2SVRule match {
+      case Some(rule) => ruleContext.isAdmin.map(if (_) rule.adminLocations else rule.assistantLocations).map(Some(_))
+      case None => Future.successful(None)
+    }
 
-      val applicableRule = biz2svRules.findOne(_.conditions.forAll(_.evaluate(authContext, ruleContext, auditContext)))
-
-      applicableRule.flatMap {
-        case Some(biz2svRule) =>
-          throttleLocation(biz2svRule).map { location =>
-            Some(twoStepVerificationThrottle.isRegistrationMandatory(biz2svRule.name, authContext.user.oid) match {
-              case true =>
-                if (isUplifted(location.mandatory)) auditContext.setSentToMandatory2SVRegister(biz2svRule.name)
-                sendAuditEvent(biz2svRule, ruleContext, mandatory = true)
-                location.mandatory
-              case _ =>
-                if (isUplifted(location.optional)) auditContext.setSentToOptional2SVRegister(biz2svRule.name)
-                sendAuditEvent(biz2svRule, ruleContext, mandatory = false)
-                location.optional
-            })
-          }
-        case _ => Future.successful(None)
+    def throttleLocation(rule: Biz2SVRule, locations: ThrottleLocations, credId: String) = twoStepVerificationThrottle.isRegistrationMandatory(rule.name, credId) match {
+      case true =>
+        if (isUplifted(locations.mandatory)) auditContext.setSentToMandatory2SVRegister(rule.name)
+        sendAuditEvent(rule, ruleContext, mandatory = true)
+        locations.mandatory
+      case _ =>
+        if (isUplifted(locations.optional)) auditContext.setSentToOptional2SVRegister(rule.name)
```

Comment:
We haven't changed any business logic. This question is out of scope of this PR. @paoloboni is best to answer. I wasn't here when uplifted location bit was implemented